common/JackEventAPI.cpp

```diff
@@ -28,24 +28,24 @@ extern "C"
 {
 #endif
 
-    LIB_EXPORT uint32_t jack_midi_get_event_count(void* port_buffer);
+    LIB_EXPORT uint32_t jack_event_get_count(void* port_buffer);
 
-    LIB_EXPORT int jack_midi_event_get(jack_midi_event_t* event,
+    LIB_EXPORT int jack_event_get(jack_event_t* event,
```

Comment:
What do you think of something like `jack_message_event` for the naming?

Reply:
Yes, the next time I rewrite this pull request I'll call it the message API instead of the event API: `jack_message_read`, `jack_message_write`. The only thing I see is that this might collide with the `JackMessageBuffer`, that handles internal log messages.

Reply:
not sure if I agree with that. why message vs event, can you elaborate?

a cool thing to do actually would be to not define the osc type (as the macro like audio and midi type) but instead leave it as generic and requesting from developers to use meta-data in order to find the type.

Reply:
I'm actually thinking of it as the message event API. There are two types of message events, an OSC and a MIDI. The prototype is a binary message.

I am lazy and haven't read all your code yet, so I don't know how this idealized model scans with what you have, please excuse me if I missed.

Reply:
I'm open for any naming suggestions.

I like the idea of a protocol-agnostic event port type, but this opens the question about backwards compatibility with existing "MIDI type" channels.

Reply:
ah, then I agree it is a good suggestion. I like the name "message event".
functions like `jack_message_event_get` make sense, at least for me..

Reply:
What about `jack_event_read_message` / `jack_event_write_message`?

Reply:
call 'em "JACK message events". The existing naming convention seems to be "subject_verb", so  `jack_message_event_read`, `jack_message_event_write`

anywhere you saw the specific `jack_midi_event` that is actually generic, the prototypical/generic name is `jack_message_event`

I haven't looked at the structs, nor am I familiar with the C conventions or this codebase's conventions, and I'm ADD'ing my way through a couple projects right now, so this is the only suggestion I have. Does it make sense?